pkg/scannerv4/client/client.go

```diff
@@ -92,28 +94,38 @@ func (c *gRPCScanner) Close() error {
 }
 
 func createGRPCConn(ctx context.Context, o connOptions) (*grpc.ClientConn, error) {
-	connOpt := clientconn.Options{
-		TLS: clientconn.TLSConfigOptions{
-			GRPCOnly:           true,
-			InsecureSkipVerify: true,
-		},
+	connOpts := []clientconn.ConnectionOption{
+		clientconn.UseInsecureNoTLS(o.skipTLS),
+		clientconn.ServerName(o.serverName),
+		clientconn.MaxMsgReceiveSize(env.ScannerV4MaxRespMsgSize.IntegerSetting()),
 	}
-	if !o.skipTLS {
-		ca, err := mtls.LoadDefaultCA()
-		if err != nil {
-			return nil, fmt.Errorf("creating CA: %w", err)
+
+	return clientconn.AuthenticatedGRPCConnection(ctx, o.address, o.mTLSSubject, connOpts...)
+}
+
+func (c *gRPCScanner) GetImageIndex(ctx context.Context, hashID string) (*v4.IndexReport, bool, error) {
+	ctx = zlog.ContextWithValues(ctx,
+		"component", "scanner/client",
+		"method", "GetImageIndex",
+		"hash_id", hashID,
+	)
+	var ir *v4.IndexReport
+	// Get the IndexReport, if it exists.
+	err := retryWithBackoff(ctx, defaultBackoff(), "indexer.GetIndexReport", func() (err error) {
+		ir, err = c.indexer.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hashID})
+		if e, ok := status.FromError(err); ok && e.Code() == codes.NotFound {
+			return nil
 		}
-		connOpt.TLS.InsecureSkipVerify = false
-		connOpt.TLS.RootCAs = ca.CertPool()
-		connOpt.TLS.UseClientCert = clientconn.MustUseClientCert
-		connOpt.TLS.ServerName = o.serverName
```

Comment:
Glad you removed this, was planning on creating a follow up item, as it didn't seem the client needed access to the root CA / key.

Reply:
client does need access to the root CA since we need to trust a non-builtin trusted CA, Central. My first iteration of this still required the CA's key, which was wrong, though. Now we don't require the key :)

Reply:
the key thing was an issue when `mtls.LoadDefaultCA()` was used in `createGRPCConn`